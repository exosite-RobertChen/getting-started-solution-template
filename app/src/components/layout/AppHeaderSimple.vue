<template>
  <v-toolbar
    color="primary"
    class="pl-2 pr-4"
    dark
    :max-height="$vuetify.breakpoint.xs || $vuetify.breakpoint.sm ? 56 : 64"
  >
    <router-link class="brandlogo-container" to="/">
      <img
        src="vendor/brandlogo.png"
        alt="brandlogo--header-simple"
        class="brandlogo"
      />
    </router-link>
    <v-spacer />
    <v-btn
      v-if="$route.name !== 'LoginPage' && !authenticated"
      :to="{ name: 'LoginPage' }"
      class="app-header__login-btn font-weight-bold"
      color="secondary"
      outlined
      router
      data-qa="HeaderLoginButton"
    >
      Login
    </v-btn>
  </v-toolbar>
</template>

<script>
export default {
  props: {
    authenticated: {
      type: Boolean,
      required: true
    }
  }
}
</script>

<style lang="scss">
.brandlogo-container {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  & > * {
    height: 45px;
  }
}
div.v-toolbar__content {
  padding: 0 !important;
}
</style>
