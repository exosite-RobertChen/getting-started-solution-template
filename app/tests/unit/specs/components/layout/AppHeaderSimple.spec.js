import Vue from 'vue'
import Vuetify from 'vuetify'
import merge from 'lodash.merge'
import { shallowMount } from '@vue/test-utils'
import AppHeaderSimple from '@/components/layout/AppHeaderSimple.vue'

const factory = overrides => {
  const localVue = Vue

  // to avoid 'component not registered' warning
  localVue.component(
    Vue.component('router-link', {
      template: `<div><slot /></div>`
    })
  )
  let defaults = {
    vuetify: new Vuetify(),
    localVue,
    propsData: {
      authenticated: false
    },
    mocks: {
      $route: {
        name: 'LandingPage'
      }
    }
  }
  merge(defaults, overrides)
  return shallowMount(AppHeaderSimple, defaults)
}

describe('AppHeaderSimple.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = factory()
  })
  afterEach(() => wrapper.destroy())

  test('can be mounted', () => {
    expect(wrapper.contains(AppHeaderSimple)).toBe(true)
  })

  test('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  describe('shows login button', () => {
    test('when not authenticated and not on login page', () => {
      expect(wrapper.contains('[data-qa=HeaderLoginButton]')).toBe(true)
    })
  })

  describe('hides login button', () => {
    test('when authenticated', () => {
      wrapper = factory({ propsData: { authenticated: true } })
      wrapper.vm.$nextTick()
      expect(wrapper.contains('[data-qa=HeaderLoginButton]')).toBe(false)
    })

    test('when on login page', () => {
      const wrapper = factory({ mocks: { $route: { name: 'LoginPage' } } })
      expect(wrapper.contains('[data-qa=HeaderLoginButton]')).toBe(false)
    })
  })
})
