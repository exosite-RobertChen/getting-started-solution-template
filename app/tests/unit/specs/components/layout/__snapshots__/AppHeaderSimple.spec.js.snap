// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeaderSimple.vue matches snapshot 1`] = `
<v-toolbar-stub
  class="pl-2 pr-4"
  color="primary"
  dark="true"
  extensionheight="48"
  maxheight="64"
  src=""
  tag="header"
  tile="true"
>
  <router-link-stub
    class="brandlogo-container"
    to="/"
  >
    <img
      alt="brandlogo--header-simple"
      class="brandlogo"
      src="vendor/brandlogo.png"
    />
  </router-link-stub>
   
  <v-spacer-stub />
   
  <v-btn-stub
    activeclass=""
    class="app-header__login-btn font-weight-bold"
    color="secondary"
    data-qa="HeaderLoginButton"
    outlined="true"
    router=""
    tag="button"
    to="[object Object]"
    type="button"
  >
    
    Login
  
  </v-btn-stub>
</v-toolbar-stub>
`;
